content-test/addon/ExperimentProvider.test.js

```diff
@@ -0,0 +1,153 @@
+    experimentProvider = new ExperimentProvider(clientID, experiments, n && (() => n));
+    experimentProvider.init();
+  }
+
+  afterEach(() => {
+    experimentProvider.destroy();
+    experimentProvider = null;
+  });
+
+  it("should have the right properties", () => {
+    setup();
+    assert.ok(experimentProvider._clientID, "should have a .clientID property");
+    assert.ok(experimentProvider._rng, "should have a ._rng property");
+    assert.ok(experimentProvider._data, "should have a ._data property");
+  });
+  it("should always generate the same number given a seed", () => {
+    setup({clientID: "foo"});
+    assert.equal(Math.round(experimentProvider._rng() * 100) / 100, 0.73, "should generate 0.73 with clientID foo");
+  });
+  it("should set .experimentId", () => {
+    setup({n: 0.8});
+    assert.equal(experimentProvider.experimentId, null, "should be null for control group");
+  });
+  it("should set .experimentId", () => {
+    setup({n: 0.1});
+    assert.equal(experimentProvider.experimentId, "foo_01", "should be foo_01 if in experiment");
+  });
+  it("should set .data ", () => {
+    setup({clientID: "baz"});
+    assert.equal(experimentProvider.data, experimentProvider._data, ".data should return return this._data");
+    assert.deepEqual(experimentProvider.data, {foo: 42}, "clientID 'baz' should result in control being picked");
+  });
+  it("should set .data", () => {
+    setup({clientID: "012j"});
+    assert.deepEqual(experimentProvider.data, {foo: 84}, "clientID '012j' should result in variant being picked");
+  });
+  it("should only select one experiment", () => {
+    const randomNumber = 0.2;
+    setup({
+      clientID: "foo",
+      experiments: {
+        kitty: {
+          name: "kitty",
+          control: {value: false},
+          variant: {id: "kitty_01", threshold: 0.2, value: true}
+        },
+        dachshund: {
+          name: "dachshund",
+          control: {value: false},
+          variant: {id: "dachshund_01", threshold: 0.2, value: true}
+        }
+      },
+      n: randomNumber
+    });
+    assert.equal(experimentProvider.data.dachshund, true, "dachshund should be selected");
+    assert.equal(experimentProvider.data.kitty, false, "kitty should not be selected");
+    assert.equal(experimentProvider.experimentId, "dachshund_01", "the experimentId should be dachshund_01");
+  });
+  it("should skip experiments with active:false", () => {
+    setup({
+      clientID: "foo",
+      experiments: {
+        foo: {
+          active: false,
+          name: "foo",
+          control: {value: "bloo"},
+          variant: {
+            id: "asdasd",
+            threshold: 0.3,
+            value: "blah"
+          }
+        }
+      }
+    });
+    assert.deepEqual(experimentProvider.data, {}, "should have empty .data");
+  });
+
+  describe("overrides", () => {
+    it("should override experiments if they are defined without setting experimentId", () => {
```

Comment:
nit: Sorry if it's just me but I read this part `without setting experimentId` as a condition when it should be a consequence. Maybe a comma? Made me go through the code twice before understanding the test so it could be a nuisance when investigating a failing test.